jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -107,15 +109,27 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
             Additional keyword arguments passed to the NDDataArray collapse operation.
             Examples include ``propagate_uncertainties`` and ``operation_ignores_mask``.
         """
+        spectral_cube, uncert_cube = None, None
         # get glue Data objects for the spectral cube and uncertainties
-        flux_viewer = self._app.get_viewer(
-            self._app._jdaviz_helper._default_flux_viewer_reference_name
-        )
-        uncert_viewer = self._app.get_viewer(
-            self._app._jdaviz_helper._default_uncert_viewer_reference_name
-        )
-        [spectral_cube] = flux_viewer.data()
-        [uncert_cube] = uncert_viewer.data()
+        for data in self.app.data_collection:
+            data_type = data.meta.get('_cube_data_type')
+
+            if data_type and data_type != 'mask':
+                if data_type == 'flux':
+                    spectral_cube = data
+                elif data_type in ['uncert', 'uncertainty']:
+                    uncert_cube = data
+
+        # verify there isn't N = 0 or N > 1 sets of cube data in data collection
+        if (spectral_cube is None) or (uncert_cube is None):
+            self.disabled_msg = "No data detected, please load data into data collection " \
+                    "and try again to compute spectral extraction."
+            return
+        elif (len([spectral_cube]) > 1) or (len([uncert_cube]) > 1):
+            self.disabled_msg = "Only one dataset is allowed in Cubeviz, please " \
+                    "remove a dataset and try again to compute spectral " \
+                    "extraction."
```

Comment:
how would this ever trigger?  Do you want to append to a list in the for-loop to raise an error if there are ever multiple matches?

How about:
```
spectral_cubes = [data for data in self.app.data_collection if data.meta.get('_cube_data_type') == 'flux']
uncert_cubes = [data for data in self.app.data_collection if data.meta.get('_cube_data_type') == 'uncert']

# any conditions to set disabled message if either list don't have length of 1, otherwise use spectral_cubes[0], uncert_cubes[0]

```

Alternatively, since we don't currently support loading extra cubes, maybe the overhead isn't worth it yet and we can just add a comment saying the logic will need to be revised to support multiple cubes?

Reply:
It wouldn't be triggered since it isn't appended to the list, its just an assignment in the loop, I do like the list comprehension better but lmk if its best to enable that or to leave the comment instead

Reply:
I think a single for-loop with the assumption (and a comment) expecting a single flux and uncertainty cube is probably the best/simplest for now.

Reply:
We don't allow loading more cubes, however you _can_ end up with multiple flux cubes in the data collection via, e.g., doing a gaussian smooth on the cube. I have more thoughts but I'll put them in a top level comment.

Reply:
that's a good point - should those be allowed as input to spectral extraction (I'm guessing not, at least for now?)

Reply:
I think in theory they should, but it may be out of scope for now.